osmosis-labs/osmosis-frontend
# 👻 Who Ya Gonna Call?: Fix referential instability in `useSwap` to stop continuous rerenders and high CPU usage

## What is the purpose of the change:

To stop the continuous rerenders and high CPU load caused by `swapState`.

![CleanShot 2024-11-16 at 15 58 16](https://github.com/user-attachments/assets/18021c91-1787-4978-85e0-2c45fa64baa8)


## Brief Changelog

- add logs for finding the leaks
- stop the leaks
- remove the log

## Testing and Verifying

This change has been tested locally by rebuilding the website and verified content and links are expected.

## Testing steps

- Checkout the first commit (640ea8edaeeba24fbcaa030e15244ed234c702ab)
- open the swap page and enter an amount 
- wait for a bit
- stop the code in the debugger and search for repeating blocks of changes
- check out the last fix commit (687f43ea34d3)
- check that the changes are gone 
- check out head and see that the logs are gone 

![CleanShot 2024-11-16 at 13 38 52](https://github.com/user-attachments/assets/07b0dc2d-9208-4233-bd02-88267f713dab)
